pinot-orc/src/main/java/org/apache/pinot/orc/data/readers/ORCRecordReader.java

```diff
@@ -58,6 +67,24 @@
 
   private static final Logger LOGGER = LoggerFactory.getLogger(ORCRecordReader.class);
 
+  @VisibleForTesting
+  public ORCRecordReader(String inputPath) {
+    Configuration conf = new Configuration();
+    try {
```

Comment:
Add a init() that takes a path instead of segment-generator config (it can be a private method). Have this constructor and the public init(SegmentGeneratorConfig) use that.